tools/ModelGeneration/Mathematica/CreateModel.m

```diff
@@ -0,0 +1,652 @@
+    }
+  }
+
+  for (std::size_t i = 0; i < NHiggs; i++)
+  {
+    for (std::size_t j = 0; j < NHiggs; j++)
+    {
+      for (std::size_t k = 0; k < NHiggs; k++)
+      {
+        TripleHiggsCorrectionsCWPhysical[i][j][k]   = 0;
+        TripleHiggsCorrectionsTreePhysical[i][j][k] = 0;
+        TripleHiggsCorrectionsCTPhysical[i][j][k]   = 0;
+        for (std::size_t l = 0; l < NHiggs; l++)
+        {
+          for (std::size_t m = 0; m < NHiggs; m++)
+          {
+            for (std::size_t n = 0; n < NHiggs; n++)
+            {
+              double RotFac =
+                  HiggsRotSort(i, l) * HiggsRotSort(j, m) * HiggsRotSort(k, n);
+              TripleHiggsCorrectionsCWPhysical[i][j][k] +=
+                  RotFac * GaugeBasis[l][m][n];
+              TripleHiggsCorrectionsTreePhysical[i][j][k] +=
+                  RotFac * LambdaHiggs_3[l][m][n];
+              TripleHiggsCorrectionsCTPhysical[i][j][k] +=
+                  RotFac * LambdaHiggs_3_CT[l][m][n];
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+void Class_Potential_"<>name<>"::SetCurvatureArrays()
+{
+  initVectors();
+  SetCurvatureDone = true;
+  for (std::size_t i = 0; i < NHiggs; i++)
+    HiggsVev[i] = vevTree[i];
+
+  // assign the non-zero entries",
+  Sequence@@Table[ "  " <> ToString[i],{i,CurvatureL1}] ,
+  Sequence@@Table[ "  " <> ToString[i],{i,CurvatureL2}] ,,
+  Sequence@@Table[ "  " <> ToString[i],{i,CurvatureL3}] ,
+  Sequence@@Table[ "  " <> ToString[i],{i,CurvatureL4}] ,,
+  Sequence@@Table[ "  " <> ToString[i],{i,GaugeCurvatureL4}] ,
+  "
+  std::complex<double> V11, V12, V13, V21, V22, V23, V31, V32, V33;
+  V11 = SMConstants.C_Vud;
+  V12 = SMConstants.C_Vus;
+  V13 = SMConstants.C_Vub;
+  V21 = SMConstants.C_Vcd;
+  V22 = SMConstants.C_Vcs;
+  V23 = SMConstants.C_Vcb;
+  V31 = SMConstants.C_Vtd;
+  V32 = SMConstants.C_Vts;
+  V33 = SMConstants.C_Vtb;
+",
+  Sequence@@Table[ "  " <> ToString[i],{i,LeptonCurvatureL3}] ,,
+  Sequence@@Table[ "  " <> ToString[i],{i,QuarkCurvatureL3}] ,"
+}
+
+bool Class_Potential_"<>name<>"::CalculateDebyeSimplified()
+{
+  return false;
+  /*
+   * Use this function if you calculated the Debye corrections to the Higgs mass
+   * matrix and implement your formula here and return true. The vector is given
+   * by DebyeHiggs[NHiggs][NHiggs]
+   */
+}
+
+bool Class_Potential_"<>name<>"::CalculateDebyeGaugeSimplified()
+{
+  /*
+   * Use this function if you calculated the Debye corrections to the gauge mass
+   * matrix and implement your formula here and return true. The vector is given
+   * by DebyeGauge[NGauge][NGauge]
+   */
+  return false;
+}
+double
+Class_Potential_"<>name<>"::VTreeSimplified(const std::vector<double> &v) const
+{
+  if (not UseVTreeSimplified) return 0;
+  double res = 0;
+  res = " <> ToC[Simplify[VHiggs/.Thread[higgsbase->Table[If[PossibleZeroQ[higgsvevFiniteTemp[[i]]],0,v[[i-1]]],{i,Length[higgsvevFiniteTemp]}]]]] <> ";
```

Comment:
With `Mathematica 11.2` this line creates a `Part::partd: Part specification v[[1]] is longer than depth of object.` error for me. `VTreeSimplified` is still produced. Any way to circumvent this error?

Reply:
This error is expected. I silenced it.